bin/s2nd.c

```diff
@@ -556,6 +573,18 @@ int main(int argc, char *const *argv)
         case 'L':
             key_log_path = optarg;
             break;
+        case 'P':
+            if (conn_settings.psk_list_len >= S2N_MAX_PSK_LIST_LENGTH) {
+                for (size_t i = 0; i < conn_settings.psk_list_len; i++) {
+                    free(conn_settings.psk_optarg_list[i]);
+                }
+                fprintf(stderr, "Error setting psks, maximum number of psks permitted is 10.\n");
+                exit(1);
+            }
+            conn_settings.psk_optarg_list[conn_settings.psk_list_len] = malloc(strlen(optarg) + 1);
+            strcpy(conn_settings.psk_optarg_list[conn_settings.psk_list_len], optarg);
```

Comment:
Is the malloc necessary here? Does the value of optarg change / become unavailable outside of this case? (I don't think so-- we don't malloc for any other optarg strings. But you'll want to read the documentation)

Reply:
Alternative to this will be using `psk_optarg_list[S2N_MAX_PSK_LIST_LENGTH[S2N_MAX_PSK_PARAMS_LENGTH]` which I think is unnecessary space.
```c
#define S2N_MAX_PSK_LIST_LENGTH 10

#define S2N_MAX_PSK_PARAMS_LENGTH                         /
              UINT16_MAX  /* Max psk identity length */   /
              + UINT16_MAX /* Max psk secret length */    /
              + 20 /* Max psk hmac length */              /        
```